multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/DetermineServiceDeleteActionsToExecuteStep.java

```diff
@@ -50,19 +51,22 @@ protected StepPhase executeStep(ProcessContext context) throws Exception {
     private StepPhase calculateDeleteActions(ProcessContext context, String serviceInstanceToDelete) {
         CloudControllerClient controllerClient = context.getControllerClient();
         CloudServiceInstance serviceInstance = controllerClient.getServiceInstance(serviceInstanceToDelete, false);
+        List<CloudServiceBinding> serviceBindings = controllerClient.getServiceAppBindings(serviceInstance.getGuid());
+
         if (serviceInstance == null) {
             getStepLogger().info(Messages.SERVICE_IS_ALREADY_DELETED, serviceInstanceToDelete);
             context.setVariable(Variables.SERVICE_DELETION_ACTIONS, Collections.emptyList());
             return StepPhase.DONE;
         }
-        if (serviceInstance.getV3Metadata() != null && isExistingService(context, serviceInstanceToDelete)) {
+        if (serviceInstance.getV3Metadata() != null
+            && (isExistingService(context, serviceInstanceToDelete) || !serviceBindings.isEmpty())) {
             context.getStepLogger()
                    .debug(Messages.WILL_ONLY_REMOVE_SERVICE_INSTANCE_METADATA_BECAUSE_THE_SERVICE_TYPE_IS_EXISTING);
             context.setVariable(Variables.SERVICE_DELETION_ACTIONS, List.of(ServiceDeletionActions.DELETE_METADATA));
             return StepPhase.DONE;
         }
-        List<CloudServiceBinding> serviceBindings = controllerClient.getServiceAppBindings(serviceInstance.getGuid());
         List<CloudServiceKey> serviceKeys = controllerClient.getServiceKeys(serviceInstance);
+
```

Comment:
I think at the end of the step you can change:
`context.setVariable(Variables.SERVICE_DELETION_ACTIONS, Collections.emptyList());`to:
`context.setVariable(Variables.SERVICE_DELETION_ACTIONS, List.of(ServiceDeletionActions.DELETE_METADATA));`

thus always deleting the metadata